test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -765,12 +765,12 @@ def __are_requirements_matching(param_dic=None, test=None):
         if (hasattr(_test, name) and
                 (getattr(_test, name, None) is not None)):
             mlist = getattr(_test, name).values()
-        if mlist:
-            for mc in mlist:
-                platform = mc.platform
-                if platform not in ['linux', 'shasta',
-                                    'cray'] and mc.hostname in _moms:
-                    _moms.remove(mc.hostname)
+            if mlist:
```

Comment:
No need for mlist any more.  Just put getattr(_test, name).values() in the for below